File: javascript_small_problems/string_and_text_processing.js/swap_case.js
function swapCase(string) {
  let result = ''
  for (let index = 0; index < string.length; index += 1) {
    if (/[a-z]/.test(string[index])) {
      result += string[index].toUpperCase();
    } else if (/[A-Z]/.test(string[index])) {
      result += string[index].toLowerCase();
    } else {
      result += string[index];
    }
  }

  return result;
}

console.log(swapCase('CamelCase'));              // "cAMELcASE"
console.log(swapCase('Tonight on XYZ-TV'));      // "tONIGHT ON xyz-tv"

File: ruby_small_problems/easy7/swap_case.rb
def swapcase(string)
  string.chars.map do |chr|
    if ("a".."z").include?(chr)
      chr.upcase
    elsif ("A".."Z").include?(chr)
      chr.downcase
    else
      chr
    end
  end.join
end

def swapcase(string)
  string.chars.map do |chr|
    ("a".."z").include?(chr) ? chr.upcase : chr.downcase
  end.join
end

# alternate

def swapcase(string)
  characters = string.chars.map do |char|
    if char =~ /[a-z]/
      char.upcase
    elsif char =~ /[A-Z]/
      char.downcase
    else
      char
    end
  end
  
  characters.join
end

p swapcase('CamelCase') == 'cAMELcASE'
p swapcase('Tonight on XYZ-TV') == 'tONIGHT ON xyz-tv'
